plugins/module_utils/lxd.py

```diff
@@ -57,10 +57,20 @@ def __init__(self, url, key_file=None, cert_file=None, debug=False):
         self.debug = debug
         self.logs = []
         if url.startswith('https:'):
-            self.cert_file = cert_file
-            self.key_file = key_file
+            try:
+                with open(cert_file, 'r') as fh:
+                    self.cert_file = cert_file
+            except FileNotFoundError:
+                raise LXDClientException('The certificate is not valid.')
+
+            try:
+                with open(key_file, 'r') as fh:
+                    self.key_file = key_file
+            except FileNotFoundError:
+                raise LXDClientException('The certificate key is not valid.')
+
             parts = generic_urlparse(urlparse(self.url))
-            ctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+            ctx = ssl._create_unverified_context(purpose=ssl.Purpose.SERVER_AUTH)
```

Comment:
Changing this is introducing a security vulnerability, since you are disabling certificate validation.

Reply:
Hi Felix,

Yes and no. By initializing LXD, the setup only asked if lxd should be reachable from the network: 
```
Would you like the LXD server to be available over the network? (yes/no) [default=no]: yes
Address to bind LXD to (not including port) [default=all]: 
Port to bind LXD to [default=8443]: 
Trust password for new clients: 
```
At this point self signed certificates for the server are generated.

There has already been a discussion on this: https://github.com/ansible-collections/community.general/issues/5616
With your PR:  #6034 i get the folowing error:
```
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:997)
```

There are changes in the ssl library between python3.9 and python3.10. Previously, the connection establishment was not checked so closely and the access was given.
Now we have two Options to get the old behavior:

```python
ctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE
```
From my point of view, however, this only disguises the fact that the remote station is not sufficiently checked, or:
```python
ctx = ssl._create_unverified_context(purpose=ssl.Purpose.SERVER_AUTH)
```

I absolutely agree with you that both solutions are unattractive, however, the change "only" restores the old behavior.

How do we continue here?

Thanks Frank

Reply:
> At this point self signed certificates for the server are generated.

Which can be OK if this is only on an internal network (or for testing), but these should be replaced by proper ones. These should not be used in production if you don't work on a trusted network (which you usually shouldn't assume, see zero trust architecture).

The correct solution is to add a new `validate_certs` option that allows to disable certificate validation. When that option is set to `true`, you can do
```py
ctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE
```
But that should happen in another PR as it is a new feature.

Reply:
Hi @felixfontein ,

we are through with the topic and now split into bug and feature? You can assign the feature to me.

Thanks Frank

Reply:
@conloos feel free to create issues or directly PR(s).